packages/tables/src/styled/StyledRow.ts

```diff
@@ -67,10 +68,24 @@ const colorStyles = (props: IStyledRowProps & ThemeProps<DefaultTheme>) => {
     background-color: ${backgroundColor};
 
     &:hover {
-      border-bottom-color: ${props.isSelected ? selectedBorderColor : hoveredBorderColor};
-      background-color: ${props.isSelected
-        ? hoveredSelectedBackgroundColor
-        : hoveredBackgroundColor};
+      border-bottom-color: ${() => {
+        if (props.isSelected) {
+          return selectedBorderColor;
+        } else if (!props.isReadOnly) {
+          return hoveredBorderColor;
+        }
+
+        return undefined;
+      }};
+      background-color: ${() => {
+        if (props.isSelected) {
+          return hoveredSelectedBackgroundColor;
+        } else if (!props.isReadOnly) {
+          return hoveredBackgroundColor;
+        }
+
+        return undefined;
+      }};
```

Comment:
Can all of this logic be pulled up above into const for `hoverBorderColor` and `hoverBackgroundColor`? It should reduce the boolean logic and make the code more readable.

Reply:
I've moved this logic into the existing conditional. Much cleaner now.